fixes for array keys; general %Bind fixes; unit tests for %Expose/%Bind

timleavitt opened the pull request:
Hi Dan,

First off, cool project!

One issue I noticed in %Expose - arrays don't include keys, which are pretty important. Here's a fix for that, plus a unit test, if you're interested.

Tim

timleavitt committed:
In %Expose, include key in projection of non-relationship arrays.

timleavitt committed:
Unit test for %Expose

dkutac commented:
Thanks Tim!

Dan

From: Tim Leavitt [mailto:notifications@github.com]
Sent: Thursday, April 13, 2017 3:57 PM
To: dkutac/cache-json-regular-vs-dynamic-object <cache-json-regular-vs-dynamic-object@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Subject: [dkutac/cache-json-regular-vs-dynamic-object] Include array keys in %Expose; unit test for %Expose (#1)


Hi Dan,

First off, cool project!

One issue I noticed in %Expose - arrays don't include keys, which are pretty important. Here's a fix for that, plus a unit test, if you're interested.

Tim

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/dkutac/cache-json-regular-vs-dynamic-object/pull/1

Commit Summary

  *   In %Expose, include key in projection of non-relationship arrays.
  *   Unit test for %Expose

File Changes

  *   A UnitTest/kutac/JSON/Adapter/Expose.cls<https://github.com/dkutac/cache-json-regular-vs-dynamic-object/pull/1/files#diff-0> (111)
  *   A UnitTest/kutac/JSON/Adapter/SamplePersistent.cls<https://github.com/dkutac/cache-json-regular-vs-dynamic-object/pull/1/files#diff-1> (59)
  *   M kutac/JSON/Adapter.cls<https://github.com/dkutac/cache-json-regular-vs-dynamic-object/pull/1/files#diff-2> (8)

Patch Links:

  *   https://github.com/dkutac/cache-json-regular-vs-dynamic-object/pull/1.patch
  *   https://github.com/dkutac/cache-json-regular-vs-dynamic-object/pull/1.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/dkutac/cache-json-regular-vs-dynamic-object/pull/1>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AJy3j0CqEN5kXJ1JclINKvfIfG8i8F3qks5rvimzgaJpZM4M8u08>.

timleavitt committed:
.gitignore/.project for working with Atelier

timleavitt committed:
%Bind - fixes, more robust error handling, unit test

Handles arrays as previously implemented for %Expose
Support for "number" type in %Bind
Correct swap of key/value in %Bind for datatype arrays
Don't ignore error %Status-es
$$ThrowStatus(<%Status>) instead of throw <%Status>
Removes debug write/break (often undesirable in event of error)

timleavitt commented:
@dkutac I've tacked on some fixes and a unit test for %Bind too. (Probably could have done that as a separate pull request, in retrospect - Git isn't exactly my strong suit.)

From a performance perspective, it'd be nice for %Bind to be code-generated at some point too. Having unit test coverage should help.
